### PR TITLE
Allow saving & sharing food recipes.

### PR DESCRIPTION
**Pull request by n1k0**
This patch partially implements a few prerequisites for this [User Story](https://www.notion.so/Introduire-une-fonction-de-comparaison-8962316a263744859a7d5917198ee3c8), allowing to save and share food recipes before we'll allow comparing them against each others in a next PR.

### CODE REVIEW

File: src/Views/Bookmark.elm [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+    | ShareTab
+
+
+view : ManagerConfig msg -> Html msg
+view ({ activeTab, switchTab } as config) =
+    div [ class "card shadow-sm" ]
+        [ div [ class "card-header" ]
+            [ [ ( SaveTab, "Sauvegarder" ), ( ShareTab, "Partager" ) ]
+                |> List.map
+                    (\( msg, label ) ->
```
Comment (magopian) [inline src/Views/Bookmark.elm:51]: how about renaming `msg` to `tab`? I'm not sure i understand the meaning of `msg` here

File: src/Views/Bookmark.elm [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+shareLinkView : ManagerConfig msg -> Html msg
+shareLinkView { session, impact, funit, copyToClipBoard, scope } =
+    let
+        shareableLink =
+            case scope of
+                Food ->
+                    Just session.queries.food
+                        |> Route.FoodBuilder impact.trigram
+                        |> Route.toString
+                        |> (++) session.clientUrl
```
Comment (magopian) [inline src/Views/Bookmark.elm:81]: Should we have a `Session.Scope` and a `Session.queryToUrl`?

**Review by magopian — COMMENTED** [review]
This is awesome! A few nits and questions that are up to you if you want to address them or not, this looks mergeable for me!

File: src/Data/Food/Builder/Recipe.elm [hunk 3]
```diff
@@ -275,7 +280,56 @@ sumMasses =
     List.map .mass >> Quantity.sum
 
 
-transformFromQuery : Db -> { a | transform : Maybe BuilderQuery.TransformQuery } -> Result String (Maybe Transform)
+toString : Recipe -> String
+toString { ingredients, transform, packaging } =
+    let
+        keepNonEmptyString string =
```
[the remark this answers is not in the document]
Comment (n1k0) [reply src/Data/Food/Builder/Recipe.elm:286]: I eventually used String.Extra.nonEmpty for that purpose 👍

File: src/Page/Food/Builder.elm [hunk 4]
```diff
@@ -53,87 +61,98 @@ type alias SelectedProcess =
...
+            session.builderDb.impacts
+                |> Impact.getDefinition trigram
+                |> Result.withDefault Impact.invalid
+
+        query =
+            maybeQuery
+                |> Maybe.withDefault queries.food
+
+        ( model, newSession, cmds ) =
+            ( { currentTime = Time.millisToPosix 0
```
[the remark this answers is not in the document]
Comment (n1k0) [reply src/Page/Food/Builder.elm:98]: Problem is taht all JSON serialization is done in Elm, so JavaScript won't have an easy access to the list of bookmarks. Updating in place bookmarks from JSON in JavaScript wouldn't look very good…

File: src/Views/Bookmark.elm (hunk 1 above)
Comment (n1k0) [reply src/Views/Bookmark.elm:51]: `SaveTab ` and `SaveTab` are member of the `Msg` type, that's why.

File: src/Views/Bookmark.elm (hunk 2 above)
Comment (n1k0) [reply src/Views/Bookmark.elm:81]: But what if we ever want to render textile and food bookmarks on a same page? 🤔

File: src/Views/Bookmark.elm (hunk 1 above)
Comment (magopian) [reply src/Views/Bookmark.elm:51]: so they are not the variants of the `ActiveTab` custom type above?

Comment (n1k0) [reply src/Views/Bookmark.elm:51]: Oooh my mistake, I was confused indeed, fixing this right now

**Review by magopian — APPROVED** [review]
All good! :+1: 🚢

**Pull request merged**